### ISSUE TITLE
Guest Attestation with AMD SEV on AKS and bonus question on SGX ?

### ISSUE THREAD

**Issue by quertenmont** [confidential-containers/documentation#87]
Dear all,

First of all, I am discovering your project on confidential containers and I am very positively surprised by all I can see here.
It looks very much like a super spreading TEE at container level with no pain for the user.

So far I've been working with AMD SEV VM node on my AKS cluster but would like to reduce the granularity and go for a VM per pod, instead of a VM per node.  So your project is definitely the best candidate for this purpose.
How mature is it for a managed K8s cluster like Azure AKS ?  Is there a priori anything that prevent me to use it there ?

Second question,  I see that you are working hard on making an attestation flow integrated with confidential containers, 
but I am wondering if I could use in the meanwhile a guest attestation from within my computer as detail here:
https://github.com/Azure/confidential-computing-cvm-guest-attestation/tree/main/cvm-attestation-sample-app

I'm already using this on my pods running on AMD-SEV Node  in my AKS cluster and works well.  I suppose I should be able to to the same with a "runtimeClassName: kata-qemu-sev" annotation on my pod...  Isn't it ?

Last question,  I played a bit with SGX and gramine gcc recently,  but I stopped because of the difficulty to make everything work in a complex application (and the performance overhead in multi process app).  
So I am also curious about the "enclave-cc" runtime.  What is it doing exactly?
Is it making SGX available from within the kata container and it's up to the code running inside the container to start an application enclaved through SGX... or is it running the entire kata container in an enclave by itself?  Obviously, I would prefer the second option, but I suppose it is not that one...  Can you enlighten to me what it actually do ?

Thanks a lot for helping,
Loic

**Comment by mythi** [comment]
> So I am also curious about the "enclave-cc" runtime. What is it doing exactly?

It's wrapping your unmodified containers into a libOS (currently Occlum, Gramine WIP) enabled SGX enclave. If your container has encrypted layers, the decryption keys are pulled from a key broker service after a successful remote attestation. Essentially enclave-cc provides the same use-case than Kata confidential containers where the TEE boundary is the pod sandbox.

> Is it making SGX available from within the kata container and it's up to the code running inside the container to start an application enclaved through SGX

This use case is supported by the official Kata releases already so no extra work is needed. enclave-cc however cannot use Kata runtime today.

**Comment by quertenmont** [comment]
so enclave-cc is comparable to "gramine gsc"  ?
One of the pain of gramine gsc (and gramine) is the manifest file needed to prepare the build.
This is not needed here ?

**Comment by mythi** [comment]
> so enclave-cc is comparable to "gramine gsc" ?

The goal of enclave-cc is to be able run unmodified containers in an enclave. For the end user this means they don't need to run `gsc` (or `occlum build/package` for that matter) for their containers before deploying them.

**Comment by quertenmont** [comment]
Thank you for the answer @mythi it's clear.

Still looking forward to get an answers on the other points... if anyone want to take his chance

**Comment by fitzthum** [comment]
I think the second part was answered in this (public) [slack thread ](https://cloud-native.slack.com/archives/C039JSH0807/p1673359559791599) so I will close for now. Please reopen if this is not resolved.

tldr; running CoCo in a nested environment is a bit tricky, although we do have the [cloud-api-adaptor](https://github.com/confidential-containers/cloud-api-adaptor) for this purpose.
